.github/workflows/nightly-rosetta-pax-build-test.yaml

```diff
@@ -31,69 +31,108 @@ permissions:
+          
           echo "BASE_LIBRARY=${{ env.BASE_LIBRARY }}" >> $GITHUB_OUTPUT
           echo "BASE_IMAGE=${BASE_IMAGE}" >> $GITHUB_OUTPUT
-          echo "PUBLISH=${{ inputs.PUBLISH }}" >> $GITHUB_OUTPUT
 
-  build:
-    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
+  amd64:
     needs: metadata
     uses: ./.github/workflows/_build_rosetta.yaml
     with:
+      ARCHITECTURE: amd64
       BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
       BASE_LIBRARY: ${{ needs.metadata.outputs.BASE_LIBRARY }}
       BASE_IMAGE: ${{ needs.metadata.outputs.BASE_IMAGE }}
     secrets: inherit
 
-  publish-build:
-    needs: [metadata, build]
-    uses: ./.github/workflows/_publish_badge.yaml
-    if: ( always() )
+  arm64:
+    needs: metadata
+    uses: ./.github/workflows/_build_rosetta.yaml
+    with:
+      ARCHITECTURE: arm64
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
+      BASE_LIBRARY: ${{ needs.metadata.outputs.BASE_LIBRARY }}
+      BASE_IMAGE: ${{ needs.metadata.outputs.BASE_IMAGE }}
     secrets: inherit
+
+  publish-mealkit:
+    needs: [metadata, amd64, arm64]
+    if: needs.metadata.outputs.PUBLISH == 'true'
+    uses: ./.github/workflows/_publish_container.yaml
     with:
-      ENDPOINT_FILENAME: 'rosetta-pax-build-status.json'
-      PUBLISH: ${{ github.event_name == 'workflow_run' || needs.metadata.outputs.PUBLISH == 'true' }}
-      SCRIPT: |
-        if [[ ${{ needs.build.result }} == "success" ]]; then
-          BADGE_COLOR=brightgreen
-          MSG=passing
-        else
-          BADGE_COLOR=red
-          MSG=failing
-        fi
-        echo "LABEL='nightly'" >> $GITHUB_OUTPUT
-        echo "MESSAGE='${MSG}'" >> $GITHUB_OUTPUT
-        echo "COLOR='${BADGE_COLOR}'" >> $GITHUB_OUTPUT
+      SOURCE_IMAGE: |
+        ${{ needs.amd64.outputs.DOCKER_TAG_MEALKIT }}
+        ${{ needs.arm64.outputs.DOCKER_TAG_MEALKIT }}
+      TARGET_IMAGE: upstream-pax
+      TARGET_TAGS: |
+        type=raw,value=mealkit,priority=500
+        type=raw,value=mealkit-${{ needs.metadata.outputs.BUILD_DATE }},priority=500
+
+  publish-final:
+    needs: [metadata, amd64, arm64]
+    if: needs.metadata.outputs.PUBLISH == 'true'
+    uses: ./.github/workflows/_publish_container.yaml
+    with:
+      SOURCE_IMAGE: |
+        ${{ needs.amd64.outputs.DOCKER_TAG_FINAL }}
+        ${{ needs.arm64.outputs.DOCKER_TAG_FINAL }}
+      TARGET_IMAGE: upstream-pax
+      TARGET_TAGS: |
+        type=raw,value=latest,priority=1000
+        type=raw,value=nightly-${{ needs.metadata.outputs.BUILD_DATE }},priority=900  
 
   test-pax:
-    needs: build
+    needs: [metadata, amd64, arm64]
     uses: ./.github/workflows/_test_pax_rosetta.yaml
     if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
     with:
       PAX_IMAGE: ${{ needs.build.outputs.DOCKER_TAGS }}
```

Comment:
`needs.amd64.outputs.DOCKER_TAG_FINAL`?

Reply:
Agree. But question arises: do we want to run this tests on arm? If yes, it would be better to split this file into 2: build and test. I created a ticket for this: https://github.com/NVIDIA/JAX-Toolbox/issues/376